cjdns-hdr/src/errors.rs

```diff
@@ -0,0 +1,12 @@
+use thiserror::Error;
+
+#[derive(Error, Copy, Clone, PartialEq, Eq, Debug)]
+pub enum HeaderError {
```

Comment:
Questionable design.

Function `parse` can only return `CannotParse` variant, though with a lot of different messages.

Similarly, fn `serialize` only returns `CannotSerialize` variant, again with a lot of different messages.

Better to split this into two distinct error types, and converte these messages into enum variants.

Reply:
took it into consideration